Firmware/mmu2_protocol_logic.cpp

```diff
@@ -126,6 +134,10 @@ void ProtocolLogic::SendMsg(RequestMsg rq) {
     uart->write(txbuff, len);
     LogRequestMsg(txbuff, len);
     RecordUARTActivity();
+    if (rq.code == RequestMsgCodes::Version && rq.value == 3 ){
```

Comment:
this piece of code doesn't belong in here - the proper way of doing this is to alter the state machines, not tweaking the SendMsg function